### PR TITLE
new package py-pyem

### PR DESCRIPTION
**Pull request by dorton21**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pyem/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class PyPyem(PythonPackage):
+    """Project-level Python virtual environment management tool.
+
+    PyEM manages multiple virtual environments local to projects.
+    It provides shortcuts to create, remove, switch between, and run
+    commands against virtual environments created against
+    various Python interpreters."""
+
+    homepage = "https://pypi.org/project/pyem/#description"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyem/package.py]: You can drop this line, it's inferred from the `pypi` attribute

File: var/spack/repos/builtin/packages/py-pyem/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    pypi     = "pyem/pyem-2.1.0.tar.gz"
+
+    depends_on('python@3.7:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+
+    version('2.1.0', sha256='5234a20427ab2813a8a0bf1e9112d4d854b1b0502b3e63d17c1b1a3c4be9340e')
+
+    def patch(self):
+        filter_file(r'from distutils.core import setup', 'from setuptools import setup', 'setup.py')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyem/package.py]: I would add a comment explaining why this patch is necessary:
```python
# distutils does not support entry_points, setuptools needed to install pyem binary
```

**Review by adamjstewart — COMMENTED** [review]
The empty line on the last line and the long line will probably cause flake8 to fail.

**Review by alalazo — CHANGES_REQUESTED** [review]
Flake8 issues:
```
var/spack/repos/builtin/packages/py-pyem/package.py:26: [E501] line too long (100 > 88 characters)
var/spack/repos/builtin/packages/py-pyem/package.py:27: [W391] blank line at end of file
```

File: var/spack/repos/builtin/packages/py-pyem/package.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    It provides shortcuts to create, remove, switch between, and run
+    commands against virtual environments created against
+    various Python interpreters."""
+
+    pypi     = "pyem/pyem-2.1.0.tar.gz"
+
+    depends_on('python@3.7:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+
+    version('2.1.0', sha256='5234a20427ab2813a8a0bf1e9112d4d854b1b0502b3e63d17c1b1a3c4be9340e')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyem/package.py]: Can you move the version above depends_on?

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**